hr_announcement/models/announcement.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class Announcement(models.Model):
+    _inherit = "announcement"
+
+    announcement_type = fields.Selection(
+        selection_add=[
+            ("employee", "Employee"),
+            ("department", "Department"),
+            ("job_position", "Job Position"),
+        ],
+        ondelete={
+            "employee": "set default",
+            "department": "set default",
+            "job_position": "set default",
+        },
+    )
+
+    employee_ids = fields.Many2many(
+        comodel_name="hr.employee",
+        relation="hr_employee_announcements",
```

Comment:
Usually we can go with the default relation name for m2m fields unless there are conflicting m2m with same relation in the same model